vividus-to-xray-for-jira-exporter/src/main/java/org/vividus/jira/context/CacheableClientCredentialsHttpClientContextFactory.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2019-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.jira.context;
+
+import org.apache.http.HttpHost;
+import org.apache.http.client.AuthCache;
+import org.apache.http.client.CredentialsProvider;
+import org.apache.http.client.protocol.HttpClientContext;
+import org.apache.http.impl.auth.BasicScheme;
+import org.apache.http.impl.client.BasicAuthCache;
+import org.vividus.http.client.ClientBuilderUtils;
+import org.vividus.jira.JiraConfiguration;
+
+public class CacheableClientCredentialsHttpClientContextFactory implements IHttpClientContextFactory
+{
+    private final CredentialsProvider credentialsProvider;
+    private final AuthCache authCache;
+
+    public CacheableClientCredentialsHttpClientContextFactory(JiraConfiguration jiraConfiguration)
```

Comment:
it should be a part of `vividus-http-client` and it's shouldn't use `JiraConfiguration`